### DISCUSSION TITLE
How to shutdown the server

### DISCUSSION THREAD

**Discussion by WangHan2014** [bluenviron/mediamtx#2886] [Questions]
## Which version are you using?

v1.4.1

## Which operating system are you using?

- [ ] Linux amd64 standard
- [ ] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [x] Other (Anroid arm64)

## Describe the issue

i'm building my own library on Android,and i want to know how to manual shutdown it but "os.exit",it'll kill whole application.

Thanks！

I'm building a AAR package for my android application using gomobile , it's work great.
First i replace the main function with StartServer().It's a similar copy of it.
![image](https://github.com/bluenviron/mediamtx/assets/53888453/827adc7a-f8a6-462e-b8e8-dec0236b4e71)

But i need a function to stop mediamtx,if i called os.Exit() ,it'll stop the whole application :-(.

And I found this in internal/core.go line 222,It look like can stop the mediamtx only not the whole application,if there are any way to triger this?.
![image](https://github.com/bluenviron/mediamtx/assets/53888453/c70b8f78-3aad-4170-9858-429717ee3e46)

Native call in JVM
![image](https://github.com/bluenviron/mediamtx/assets/53888453/7cfd9cd5-aca9-47b9-94e9-dfccc50263f3)

## Describe how to replicate the issue

1. start the server
2. publish with ...
3. read with ...

## Did you attach the server logs?

no

## Did you attach a network dump?

no

**Comment by aler9** [comment]
Hello, first of all remove `s.Wait()` from `StartServer()`, since it causes the function to never return.

Then add `s.Close()` inside `StopServer()`.

Pseudocode:

```go
var s core.Core

func StartServer() {
    s, ok = core.New()
    ...
}

func StopServer() {
    s.Close()
}
```

[a comment by an automated account is not shown]
